csrc/id_model/indexing.cpp

```diff
@@ -113,32 +93,33 @@ std::tuple<std::vector<IterDomain*>, std::vector<Val*>> getAllocationDomains(
       allocation_domains = tv->getLogicalDomain();
       contiguity = tv->domain()->contiguity();
     } else {
-      const auto inlining_pos = tv->getComputeAtPosition();
```

Comment:
In this function, currently there are multiple check points for which ID to skip. I feel that this is confusing, and indeed, it could miss IDs for some case. For example, if `tv->hasAllocation()` is true, then we will not remove grid parallelized IDs in the allocation domain if `tv` is smem tensor. I refactored this function to have a single check point regarding which ID to skip.